core/domain/exp_domain.py

```diff
@@ -1343,8 +1348,7 @@ def validate(self, exp_param_specs_dict):
 
         if self.hints:
             if self.solution:
-                Solution.from_dict(
-                    self.id, self.solution).validate(self.id)
+                self.solution.validate(self.id)
```

Comment:
Yes, so then I think the check is in the right place. It is in the validate() method inside the InteractionInstance Object.